src/components/panel-body/UEpage/communication/form/CommunicationRequestForm.jsx

```diff
@@ -0,0 +1,155 @@
+      valueName: 'type',
+      values: typeRadioValues,
+      previousLabel: D.previousButton,
+      nextLabel: D.confirmButton,
+    },
+    {
+      title: D.selectCommunciationRequestReason,
+      valueName: 'reason',
+      values: reasonRadioValues,
+      previousLabel: D.previousButton,
+      nextLabel: D.confirmButton,
+    },
+    {
+      title: D.communicationRequestValidation,
+      previousLabel: D.previousButton,
+      nextLabel: D.sendButton,
+    },
+  ];
+  const [selectedStep, setSelectedStep] = useState(steps[0]);
+
+  const [currentAttributeIsValid, setCurrentAttributeIsValid] = useState(false);
+
+  const getStepIndex = () => steps.findIndex(step => step.title === selectedStep.title);
+  const isFirstStep = () => getStepIndex() === 0;
+  const isLastStep = () => getStepIndex() === steps.length - 1;
+
+  const generateCommunicationRequest = commRequ => ({
+    ...commRequ,
+    status: [{ date: new Date().getTime(), status: communicationStatusEnum.INIT.type }],
+  });
+
+  const nextStepAction = () => {
+    if (isLastStep()) {
+      saveFunction(generateCommunicationRequest(communicationRequest));
+      closeModalFunction();
+    } else {
+      setSelectedStep(steps[getStepIndex() + 1]);
+    }
+  };
+
+  const previousStepAction = () => {
+    if (isFirstStep()) {
+      closeModalFunction();
+    } else {
+      setSelectedStep(steps[getStepIndex() - 1]);
+    }
+  };
+
+  const isValueValid = value => !!value;
+
+  const isAddressingValid = () => {
+    // TODO : add business rules here
+    return true;
+  };
+
+  const checkValidity = useCallback((communicationRequest, selectedStep) => {
+    if (selectedStep.valueName !== undefined)
+      return isValueValid(communicationRequest[selectedStep.valueName]);
+
+    return isAddressingValid();
+  }, []);
+
+  useEffect(() => {
+    setCurrentAttributeIsValid(checkValidity(communicationRequest, selectedStep));
+  }, [checkValidity, communicationRequest, selectedStep]);
+
+  const handleChange = event => {
+    const {
+      target: { value },
+    } = event;
+    setCommunicationRequest({ ...communicationRequest, [selectedStep.valueName]: value });
+  };
+  const addressesErrors = checkCommunicationRequestFormAddressesValidity(
+    recipientInformation,
+    userInformation,
+    communicationRequest
+  );
+
+  return (
+    <>
+      <Typography className={classes.title}>{selectedStep.title}</Typography>
+      {selectedStep.valueName && (
+        <FormRadioGroup
+          groupSelectedValue={communicationRequest[selectedStep.valueName]}
+          groupHandleChange={handleChange}
+          groupValues={selectedStep.values}
+        />
+      )}
+      {selectedStep.valueName === undefined && (
+        <CommunicationRequestValidation
+          communicationRequest={communicationRequest}
+          recipientInformation={recipientInformation}
+          userInformation={userInformation}
+          addressesErrors={addressesErrors}
+        />
+      )}
```

Comment:
I think (and sonar too) it's cleaner to use ternary conditional operator in this case